src/dotnet/APIView/ClientSPA/src/app/_components/shared/comment-thread/comment-thread.component.spec.ts

```diff
@@ -28,4 +29,39 @@ describe('CommentThreadComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  describe('setCommentResolutionState', () => {
+    it ('should select latest user to resolve comment thread', () => {
+      const comment1 = {
+        id: '1',
+        isResolved: true,
+        changeHistory: [ {
+          changeAction: 'resolved', 
+          changedBy: 'test user 1',
+        },
+        {
+          changeAction: 'resolved', 
+          changedBy: 'test user 2',
+        }]
+      } as CommentItemModel;
+      const comment2 = {
+        id: '2',
+        isResolved: true,
+        changeHistory: [ {
+          changeAction: 'resolved', 
+          changedBy: 'test user 1',
+        },
+        {
+          changeAction: 'resolved', 
+          changedBy: 'test user 2',
+        }]
+      } as CommentItemModel;
+      
+      component.codePanelRowData!.comments = [comment1, comment2];
+      component.codePanelRowData!.isResolvedCommentThread = true;
+      fixture.detectChanges();
+      component.setCommentResolutionState();
+      expect(component.threadResolvedBy).toBe('test user 2');
```

Comment:
not sure I follow the tests (it could be because of my misunderstanding of the `changeHistory` property.
Both `comment` have the same properties except the `id` . Could you please explain?

Reply:
The change history has latest changes below. So this is just making sure the latest changes are used.

Reply:
Ahhh I see. Yeah the update you made to the test makes sense.